spec/controllers/reports_controller_spec.rb

```diff
@@ -66,5 +66,15 @@
         expect(assigns(:report_data).count).to eql(1)
       end
     end
+
+    describe 'GET #public' do
```

Comment:
I wonder if we should test that the `@data` variable is assigned. 

I usually use double to verify the right object is being assigned, but I'm not saying it's the best option:

```ruby
let(:report_view) { double } 

before do
  allow(Views::Reports::PublicSubmissionData).to receive(:new).and_return(report_view)
end

it 'assigns the report view as @data' do
  expect(assigns(:data)).to eql(report_view)
end
```

Something between those lines.